gapic/templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2

```diff
@@ -178,6 +179,49 @@ def test__get_api_endpoint():
         {{ service.client_name }}._get_api_endpoint(None, mock_client_cert_source, mock_universe, "auto")
     assert str(excinfo.value) == "mTLS is not supported in any universe other than googleapis.com."
 
+{% if service.version %}
+{% for method in service.methods.values() %}{% with method_name = method.name|snake_case %}
+{% for mode in ["", "async"] %}
+{% if mode == "async" %}
+async def test_{{ method_name }}_api_version_header_async(transport_name="grpc"):
+    client = {{ service.async_client_name }}(credentials=ga_credentials.AnonymousCredentials(), transport=transport_name)
+{% else %}
+@pytest.mark.parametrize("transport_name", [
+    {% if 'grpc' in opts.transport %}
+    ("grpc"),
+    {% endif %}
+    {% if 'rest' in opts.transport %}
+    ("rest"),
+    {% endif %}
+])
+def test_{{ method_name }}_api_version_header(transport_name):
+    client = {{ service.client_name }}(credentials=ga_credentials.AnonymousCredentials(), transport=transport_name)
+{% endif %}
+    # TODO: Make this test unconditional once the minimum supported version of
+    # google-api-core becomes 2.19.0 or higher.
+    api_core_major, api_core_minor = [int(part) for part in api_core_version.__version__.split(".")[0:2]]
+    if api_core_major > 2 or (api_core_major == 2 and api_core_minor >= 19):
```

Comment:
We need to guard our tests like this every time a new feature is dependent on a version of `api-core` which is not the minimum version. This will also add a lot of toil for future clean up.

Instead, we should decide a generalized way of doing this and also keep track of the things that we need to clean up.

I would suggest using a helper function as a decorator for this. Something like:

```python
def is_supported_version(major_version, minor_version):
    def decorator(func):
        @wraps(func)
        def wrapper(*args, **kwargs):
            api_core_major, api_core_minor = [int(part) for part in api_core_version.__version__.split(".")[0:2]]
            if api_core_major > major_version or (api_core_major == major_version and api_core_minor >= minor_version):
                return func(*args, **kwargs)
            else:
               print("Skipping test due to unsupported version ...")
        return wrapper
    return decorator
```

Let me know what you think!

Reply:
A good idea. See my related [comment](https://github.com/googleapis/gapic-generator-python/pull/1999#discussion_r1586576476) about being able to just have a helper function to get the version parts. The decorator could be done as part of that work.

Reply:
I filed https://github.com/googleapis/gapic-generator-python/issues/2023 to clean up the function as a follow up task so we don't block this PR indefinitely with improvements